packages/testing/src/index.ts

```diff
@@ -0,0 +1,197 @@
+      "Content-Type": "application/json",
+      "x-trigger-api-key": apiKey,
+      "x-trigger-action": action,
+      "x-trigger-endpoint-id": opts.endpointId,
+      "x-trigger-job-id": opts.jobId,
+    },
+    body: JSON.stringify(opts.body),
+  });
+};
+
+const buildRequestBody = (event: RunJobBody["event"], job: RunJobBody["job"]): RunJobBody => ({
+  event,
+  job,
+  run: {
+    id: String(Math.random()),
+    isTest: false,
+    isRetry: false,
+    startedAt: new Date(),
+  },
+  environment: {
+    id: String(Math.random()),
+    slug: "test-env",
+    type: "DEVELOPMENT",
+  },
+  organization: {
+    id: String(Math.random()),
+    title: "Test Org",
+    slug: "test-org",
+  },
+});
+
+export const createJobTester =
+  (vi: typeof vitestVi) =>
+  async <
+    TTrigger extends Trigger<EventSpecification<any>>,
+    TIntegrations extends Record<string, TriggerIntegration<IntegrationClient<any, any>>> = {},
+  >(
+    job: Job<TTrigger, TIntegrations>,
+    opts: {
+      payload?: TriggerEventType<TTrigger>;
+      tasks?: Record<string, any>;
+    } = {}
+  ): Promise<
+    {
+      io: IOWithIntegrations<TIntegrations>;
+      status: string;
+      output: any;
+    } & Record<string, any>
+  > => {
+    const mockSendEvent = (client: TriggerClient) =>
+      vi
+        .spyOn(client, "sendEvent")
+        .mockImplementation(
+          async (unparsedEvent: SendEvent, unparsedOptions: SendEventOptions = {}) => {
+            const body = SendEventBodySchema.parse({
+              event: unparsedEvent,
+              options: unparsedOptions,
+            });
+            const { event, options } = body;
+
+            const timestamp = new Date();
+            const deliverAt =
+              options?.deliverAt ||
+              (options?.deliverAfter
+                ? new Date(Date.now() + options.deliverAfter * 1000)
+                : undefined);
+
+            const eventLog = {
+              id: event.id,
+              name: event.name,
+              payload: event.payload ?? {},
+              context: event.context,
+              timestamp,
+              deliverAt,
+              deliveredAt: !deliverAt ? timestamp : undefined,
+              cancelledAt: undefined,
+            };
+
+            return eventLog;
+          }
+        );
+
+    const { client, trigger } = job;
+
+    mockSendEvent(client);
+    const eventLog = await client.sendEvent({
+      name: typeof trigger.event.name === "string" ? trigger.event.name : trigger.event.name[0],
+      payload: opts.payload,
+    });
+
+    const { integrations, run } = job.options;
+    let ioSpy: IOWithIntegrations<TIntegrations> | undefined;
+
+    const taskMockImplementation = async (key: string) => {
+      if (!opts.tasks || typeof key !== "string") return;
+      return opts.tasks[key];
+    };
+
+    job.options.run = (payload, io, ctx) => {
+      if (integrations) {
+        for (const [integrationName, integration] of entries(integrations)) {
```

Comment:
Because we slightly changed how integrations work a few days ago, this unfortunately doesn't work anymore. I took a quick look to see if I could fix it easily but I couldn't. 

Tasks are now just methods inside integrations, and can even be nested inside other objects.

Given that they're all powered by io.runTask, we could mock that function. This doesn't work but is the rough idea:

```ts
job.options.run = (payload, io, ctx) => {
      vi.spyOn(io, "runTask").mockImplementation((key, callback) => {
        //do some mocked thing instead
      });

      if (integrations) {
        for (const [integrationName, integration] of entries(integrations)) {
          const readyIntegration = integration.cloneForRun(io, integrationName as string);

          io[integrationName] = readyIntegration;
        }
      }
      return run(payload, io, ctx);
    };
```